front/lib/api/assistant/conversation/attachments.ts

```diff
@@ -91,17 +91,25 @@ export function conversationAttachmentId(
 export function getAttachmentFromContentFragment(
   cf: ContentFragmentType
 ): ConversationAttachmentType | null {
-  if (isContentNodeContentFragment(cf)) {
-    return getAttachmentFromContentNodeContentFragment(cf);
+  // Expired content fragments cannot be converted to attachments
+  if (cf.expiredReason) {
+    return null;
+  }
+
+  // After this check, we know expiredReason is null
+  const nonExpiredCf = cf as ContentFragmentType & { expiredReason: null };
```

Comment:
Is this actually needed ? I would expect type narrowing to just work.

If not, I think we should make this a type guard, because `cf as ContentFragmentType & { expiredReason: null };` is a bit against our code rules